src/instructlab/model/backends/backends.py

```diff
@@ -95,12 +95,24 @@ def is_model_gguf(model_path: pathlib.Path) -> bool:
         with model_path.open("rb") as f:
             first_four_bytes = f.read(4)
 
-        # Convert the first four bytes to an integer
-        first_four_bytes_int = int(struct.unpack("<I", first_four_bytes)[0])
+        try:
+            # Convert the first four bytes to an integer
+            first_four_bytes_int = int(struct.unpack("<I", first_four_bytes)[0])
+        except struct.error as exc:
+            logger.debug(
+                f"Failed to unpack the first four bytes of {model_path}. "
+                f"The file might not be a valid GGUF file or is corrupted: {exc}"
+            )
+            return False
 
         return first_four_bytes_int == GGUF_MAGIC
     except IsADirectoryError as exc:
-        logger.debug(f"GGUF Path is a directory, returning {exc}")
+        logger.debug(f"GGUF Path {model_path} is a directory, returning {exc}")
+        return False
+    except OSError as exc:
```

Comment:
Sorry I just realized that the inner try/except is not needed, given that we are already in a try/except block you can just append this in here:

```
        except struct.error as exc:
            logger.debug(
                f"Failed to unpack the first four bytes of {model_path}. "
                f"The file might not be a valid GGUF file or is corrupted: {exc}"
            )
```

Thanks!